--- source:Inner::refcount
+++ verus:Inner::refcount
@@ -1,17 +1,26 @@
     fn refcount(&self, frame: FrameAddress) -> Result<u8, Error> {
-        let frame_number: usize = frame.into_frame_number().into_raw_value();
+        
+        
+        let raw: usize = frame.into_raw_value();
+        let frame_number: usize = raw / mem::FRAME_SIZE;
+        
 
         if frame_number >= self.refcount.len() {
+            
             let reason: &str = "frame number out of bounds";
+            
             error!("{reason} (frame={frame:?})");
             return Err(Error::new(ErrorCode::BadAddress, reason));
         }
 
         if self.refcount[frame_number] == 0 {
+            
             let reason: &str = "frame is not allocated";
+            
             error!("{reason} (frame={frame:?})");
             return Err(Error::new(ErrorCode::BadAddress, reason));
         }
 
+        
         Ok(self.refcount[frame_number])
     }
